cmd/state/internal/cmdtree/ppm.go

```diff
@@ -171,25 +177,79 @@ func addOtherCommands(cmds []*captain.Command) []*captain.Command {
 			"config",
 			"configuration settings",
 			func(_ *captain.Command, _ []string) error {
-				return printDefault()
+				return printDefault(prime)
 			},
 		),
 		captain.NewShimCommand(
 			"gui",
 			"opens the graphical user-interface",
 			func(_ *captain.Command, _ []string) error {
-				return printDefault()
+				return printDefault(prime)
 			},
 		),
 	)
 }
 
-func addInfoCommand(cmds []*captain.Command) []*captain.Command {
+func addInfoCommand(prime *primer.Values, cmds []*captain.Command) []*captain.Command {
 	return append(cmds, captain.NewShimCommand(
 		"info",
 		"prints ppm help message",
 		func(_ *captain.Command, _ []string) error {
-			return printMain()
+			return printMain(prime)
 		},
 	))
 }
+
+func shim(prime *primer.Values, intercepted, replaced, localeID string, args ...string) error {
+	err := shim(prime, intercepted, replaced, localeID, args...)
+	if err != nil {
+		analytics.EventWithLabel(analytics.CatPPMShimCmd, "error", errs.Join(err, " :: ").Error())
+	} else {
+		analytics.EventWithLabel(analytics.CatPPMShimCmd, "success", fmt.Sprintf("intercepted=%s, replaced=%s", intercepted, replaced))
+	}
+	return err
+}
+
+func executeShim(prime *primer.Values, intercepted, replaced, localeID string, args ...string) error {
+	pj, fail := projectfile.GetSafe()
+	if fail != nil && !fail.Type.Matches(projectfile.FailNoProject) {
+		return locale.WrapError(fail.ToError(), "err_ppm_get_projectfile", "Encountered unexpected error loading projectfile")
+	}
+	if pj == nil {
+		// TODO: Replace this function call when conversion flow is complete
+		analytics.Event(analytics.CatPPMShimCmd, "tutorial")
+		return tutorial()
+	}
+
+	commands := strings.Split(replaced, " ")
+	replacedArgs := args
+	if len(commands) > 1 {
+		replaced = commands[0]
+		replacedArgs = commands[1:]
+		replacedArgs = append(replacedArgs, args...)
+	}
+
+	forwarded := []string{"state", replaced}
+	forwarded = append(forwarded, replacedArgs...)
+	prime.Output().Print(locale.Tr(localeID, strings.Join(forwarded, " "), intercepted))
+
+	return invoke(replaced, replacedArgs...)
+}
+
+func invoke(command string, args ...string) error {
```

Comment:
Please be sure to replace this with my code once my PR is merged (before merging this one).